src/main.cpp

```diff
@@ -2516,7 +2516,8 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             return AbortNode(state, "Failed to write transaction index");
 
     // add this block to the view's block chain
-    view.SetBestBlock(pindex->GetBlockHash());
+    if (!fJustCheck)
+        view.SetBestBlock(pindex->GetBlockHash());
```

Comment:
No need for this. As if `fJustCheck` is true, we've already exited the function by now (line 2458).

Reply:
yeah good catch. Just dropped the last commit.